From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brad Miller <brad@bradhouse.com>
Date: Sat, 15 Nov 2014 19:19:36 -0500
Subject: [PATCH 0484/6262] Using netconsole-host to start programs and get
 output directed to netconsole and the log file (fixes artf3777, artf3750)

Change-Id: I904989f94132dd0031b5bec9cca2c8dee46070fe
---
 .../src/main/resources/cpp-zip/ant/robotCommand             | 2 +-
 hal/lib/Athena/HAL.cpp                                      | 6 ++++++
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp/src/main/resources/cpp-zip/ant/robotCommand b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp/src/main/resources/cpp-zip/ant/robotCommand
index 85b3b5689af4c15d7c5ac7f29ad538658de4f791..06757bd83de6cd6fb699a4cdf9c928e6326c7701 100644
--- a/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp/src/main/resources/cpp-zip/ant/robotCommand
+++ b/eclipse-plugins/edu.wpi.first.wpilib.plugins.cpp/src/main/resources/cpp-zip/ant/robotCommand
@@ -1 +1 @@
-/home/lvuser/FRCUserProgram
+/usr/local/frc/bin/netconsole-host /home/lvuser/FRCUserProgram
diff --git a/hal/lib/Athena/HAL.cpp b/hal/lib/Athena/HAL.cpp
index 8512c1ae8632b629425ef1341220a75e2c387232..7b4774535874eadc8a249432741be07f0128cd6d 100644
--- a/hal/lib/Athena/HAL.cpp
+++ b/hal/lib/Athena/HAL.cpp
@@ -11,6 +11,7 @@
 #include <fstream>
 #include <iostream>
 #include <unistd.h>
+#include <sys/prctl.h>
 #include <signal.h> // linux for kill
 const uint32_t solenoid_kNumDO7_0Elements = 8;
 const uint32_t dio_kNumSystems = tDIO::kNumSystems;
@@ -214,6 +215,11 @@ void HALSetNewDataSem(pthread_mutex_t * param)
  */
 int HALInitialize(int mode)
 {
+    setlinebuf(stdin);
+    setlinebuf(stdout);
+
+    prctl(PR_SET_PDEATHSIG, SIGTERM);
+
 	FRC_NetworkCommunication_Reserve(nullptr);
 	// image 4; Fixes errors caused by multiple processes. Talk to NI about this
 	nFPGA::nRoboRIO_FPGANamespace::g_currentTargetClass =
